.github/workflows/main.yml

```diff
@@ -0,0 +1,33 @@
+name: Build docker images
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+    steps:
+      # Checks out repository under $GITHUB_WORKSPACE, so we can access it
+      - uses: actions/checkout@v2
+
+      # Runs build script
+      - name: Run build script
+        run: |
+          # always push
+          export PUSH=true
+          # if this is a push to master and not a PR, also push latest - GITHUB_HEAD_REF is only set for PRs
+          if [ -z ${GITHUB_HEAD_REF+x} ] && [ "$GITHUB_REF" == "refs/heads/master" ]; then export PUSH=latest; fi
+          # log into quay.io so we can push
+          echo "$QUAY_PASSWORD" | docker login quay.io -u "$QUAY_USER" --password-stdin
```

Comment:
We've just moved our image repository to github's packages repo, so that's what we need to log into here. IIUC the github actions runner comes with some credentials, but I'll need to work out how to grant those credentials access to the right docker image repository.